kernel/main.c

```diff
@@ -10,12 +10,13 @@ volatile static int started = 0;
 void
 main()
 {
+  // Initialize an array to hold a waitcycle count per cpu.
+  uint waitcycle[NCPU];
```

Comment:
Do we need this to be an array?

Since, if I understand it correctly, main() is run by every cpu core independently, each with its own stack. If we put waitcycle on the stack, there is no need for this to be an array.

Reply:
same asking